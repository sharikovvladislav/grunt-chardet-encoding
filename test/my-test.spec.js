const { executeCommand } = require('./helpers/common');

jest.setTimeout(60000);

const TEST_PROJECT_DIR = 'test/test-project';

test('test utf8 target', async () => {
  await executeCommand('npm run clear', TEST_PROJECT_DIR);
  await executeCommand('npm i', TEST_PROJECT_DIR);
  const result = await executeCommand('npx grunt encoding:utf8', TEST_PROJECT_DIR);

  expect(result).toMatchSnapshot();
});

test('test win1251 target', async () => {
  await executeCommand('npm run clear', TEST_PROJECT_DIR);
  await executeCommand('npm i', TEST_PROJECT_DIR);
  const result = await executeCommand('npx grunt encoding:win1251', TEST_PROJECT_DIR);

  expect(result).toMatchSnapshot();
});

function kek () {
  return new Promise(resolve => setTimeout(resolve, 3000))
}