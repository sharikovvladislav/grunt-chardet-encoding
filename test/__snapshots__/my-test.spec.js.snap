// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test utf8 target 1`] = `
"Running \\"encoding:utf8\\" (encoding) task
the file utf8-test/utf8-first.js was detected like UTF-8 encoding 
the file utf8-test/utf8-second.txt was detected like UTF-8 encoding 
the file utf8-test/win1251-second.js was detected like windows-1251 encoding 
the file utf8-test/win1251.txt was detected like windows-1251 encoding 
Warning: 

Problems: 
The encoding of file utf8-test/win1251-second.js is \\"windows-1251\\", but it has to be in [UTF-8]
The encoding of file utf8-test/win1251.txt is \\"windows-1251\\", but it has to be in [UTF-8]
 Use --force to continue.

Aborted due to warnings.
"
`;

exports[`test win1251 target 1`] = `
"Running \\"encoding:win1251\\" (encoding) task
the file win1251-test/win1251-another.txt was detected like windows-1251 encoding 
the file win1251-test/win1251-third.txt was detected like windows-1251 encoding 
the file win1251-test/win1251.txt was detected like windows-1251 encoding 

Done.
"
`;
